server/graphql/v2/mutation/MemberInvitationMutations.js

```diff
@@ -0,0 +1,155 @@
+      account: {
+        type: GraphQLNonNull(AccountReferenceInput),
+        description: 'Reference to an account for the inviting Collective',
+      },
+      role: {
+        type: GraphQLNonNull(MemberRole),
+        description: 'Role of the invitee',
+      },
+      description: {
+        type: GraphQLString,
+      },
+      since: {
+        type: GraphQLDateTime,
+      },
+    },
+    async resolve(_, args, req) {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be logged in to invite a member.');
+      }
+
+      let { memberAccount, account } = args;
+
+      memberAccount = await fetchAccountWithReference(memberAccount, { throwIfMissing: true });
+      account = await fetchAccountWithReference(account, { throwIfMissing: true });
+
+      if (!req.remoteUser.isAdminOfCollective(account)) {
+        throw new Unauthorized('Only admins can send an invitation.');
+      }
+
+      if (![MemberRoles.ACCOUNTANT, MemberRoles.ADMIN, MemberRoles.MEMBER].includes(args.role)) {
+        throw new Forbidden('You can only invite accountants, admins, or members.');
+      }
+
+      const memberParams = {
+        ...pick(args, ['role', 'description', 'since']),
+        MemberCollectiveId: memberAccount.id,
+        CreatedByUserId: req.remoteUser.id,
+      };
+
+      // Invite member
+      return models.MemberInvitation.invite(account, memberParams);
+    },
+  },
+  editMemberInvitation: {
+    type: MemberInvitation,
+    description: 'Edit an existing member invitation of the Collective',
+    args: {
+      memberAccount: {
+        type: GraphQLNonNull(AccountReferenceInput),
+        description: 'Reference to an account for the member to edit.',
+      },
+      account: {
+        type: GraphQLNonNull(AccountReferenceInput),
+        description: 'Reference to an account for the Collective',
+      },
+      role: {
+        type: MemberRole,
+        description: 'Role of member',
+      },
+      description: {
+        type: GraphQLString,
+      },
+      since: {
+        type: GraphQLDateTime,
+      },
+    },
+    async resolve(_, args, req) {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be logged in to invite a member.');
+      }
+
+      let { memberAccount, account } = args;
+
+      memberAccount = await fetchAccountWithReference(memberAccount, { throwIfMissing: true });
+      account = await fetchAccountWithReference(account, { throwIfMissing: true });
+
+      if (!req.remoteUser.isAdminOfCollective(account)) {
+        throw new Unauthorized('Only admins can edit members.');
+      }
+
+      if (![MemberRoles.ACCOUNTANT, MemberRoles.ADMIN, MemberRoles.MEMBER].includes(args.role)) {
+        throw new Forbidden('You can only edit accountants, admins, or members.');
+      }
+
+      // Edit member invitation
+      const editableAttributes = pick(args, ['role', 'description', 'since']);
+
+      return models.MemberInvitation.update(editableAttributes, {
```

Comment:
It seems to me that `models.Xxx.update` returns the list of updated entries. Is the return value ok here?

Reply:
hmm not ok, I think we will need to re-fetch with `.findOne`